kartothek/core/types.py

```diff
@@ -0,0 +1,14 @@
+from typing import Callable
+
+from simplekv import KeyValueStore
+
+from kartothek.serialization import (
+    ConjunctionType,
+    LiteralType,
+    LiteralValue,
+    PredicatesType,
+)
+
+StoreFactory = Callable[[], KeyValueStore]
+
+__all__ = ["ConjunctionType", "LiteralType", "LiteralValue", "PredicatesType"]
```

Comment:
Not sure how we should deal with this. The store factory is something I encountered very frequently and the `PredicatesType` as well. Maybe it would make sense to define a type here for every kwarg we expose?

Reply:
I am ok with the common complex types, but I think we should avoid the indirection if it is not really needed (like `Optional[str]` is not too complex IMHO).